src/opnsense/mvc/app/views/OPNsense/Diagnostics/fw_log.volt

```diff
@@ -313,7 +321,8 @@
                     apply_filter();
 
                     // limit output, try to keep max X records on screen.
-                    $("#grid-log > tbody > tr:gt("+max_rows+")").remove();
+                    $(".log_tr").filter(":hidden").remove();
+                    $(".log_tr").slice(max_rows).remove();
```

Comment:
@kulikov-a if you are going to remove all outside of visible range, you might need to reset the grid on filter changes as well. Not sure if we are going to implement this, but I certainly like the `$.Deferred()` in `fetch_log()`. The `grid_reset` might be avoided by  setting `last_digest` to `false` by the way, saves a variable which more or less has the same function anyway.